JsonSchema.Generation/Generators/ObjectSchemaGenerator.cs

```diff
@@ -68,10 +68,7 @@ public void AddConstraints(SchemaGenerationContextBase context)
 
 			var memberContext = SchemaGenerationContextCache.Get(member.GetMemberType(), unconditionalAttributes);
 
-			var name = SchemaGeneratorConfiguration.Current.PropertyNamingMethod(member.Name);
-			var nameAttribute = unconditionalAttributes.OfType<JsonPropertyNameAttribute>().FirstOrDefault();
-			if (nameAttribute != null)
-				name = nameAttribute.Name;
+			var name = SchemaGeneratorConfiguration.Current.PropertyNameResolver(member);
```

Comment:
This the most significant change. It must be seen together with the implementation of the consumed property.